src/compileModel.ts

```diff
@@ -65,7 +65,7 @@ const compileModel = <TSchema extends GenericObject = GenericObject>(
 		public readonly __v: string | null;
 
 		/** Private id tracking property */
-		private __id: string | null;
+		#__id: string | null;
```

Comment:
Is this the only place where we were prefixing private members with an underscore? With the private keyword it doesn't seem necessary anymore.

Reply:
I agree about not needing the underscore prefix any longer, but we probably have to be a little careful with `Model`/`Document` instances since the properties of the schema get merged into it.  I suppose no identifier is truly safe from being used by a consumer (short of throwing an error in the `Schema` constructor if they get used), but hopefully something prefixed with two underscores will avoid a potential clash.

I believe I removed the underscore from everywhere otherwise, with the exception of `_id` , `__v`, and `_raw` which are named as such by convention.

Reply:
The nice thing about the use of the private identifier is that `this.#_id` and `this._id` can coexist so it can't be accidentally clobbered.

Reply:
Hmm, I think I knew that, but didn't really think about it.  I will convert it over in my other open PR.